pkg/controllers/jobset_controller.go

```diff
@@ -746,6 +746,12 @@ func labelAndAnnotateObject(obj metav1.Object, js *jobset.JobSet, rjob *jobset.R
 	annotations[jobset.JobIndexKey] = strconv.Itoa(jobIdx)
 	annotations[jobset.JobKey] = jobHashKey(js.Namespace, jobName)
 
+	// Apply coordinator annotation/label if a coordinator is defined in the JobSet spec.
+	if js.Spec.Coordinator != nil {
+		labels[jobset.CoordinatorKey] = coordinatorEndpoint(js)
```

Comment:
One subtle bug here would be that if someone specifies these labels we would overwrite.

Do we want a warning if someone uses these labels?

Reply:
Yep, I have a follow up PR with webhook changes ready to go after this is merged. Will tag you on it shortly.